### PR TITLE
Do not merge this -- tell me what you think

### PR DESCRIPTION
**Pull request by karth295**

### CODE REVIEW

**Comment by karth295** [comment]
I tested that this patch works

File: yaas/client.py [hunk 1]
```diff
@@ -36,11 +37,33 @@ def __init__(
         self.raw = raw
         self.debug = debug
 
-        self.blueprint = Blueprint(self)
         self.cluster = Cluster(self)
         self.host = Host(self)
         self.repo = Repo(self)
 
+    def blueprint_add(self, blueprint_name, blueprint, validate_topology=True):
```
Comment (dmtucker) [inline yaas/client.py:42]: Maybe this should be boolean?

File: yaas/client.py [hunk 2]
```diff
@@ -36,11 +37,33 @@ def __init__(
...
+            '/api/v1/blueprints/{name}'.format(name=blueprint_name),
+            data=json.dumps(blueprint.serialize()),  # `json=blueprint` would be better, but it doesn't work!
+            params={'validate_topology': validate_topology})
+
+    def blueprint_list(self):
+        response = self.request('get', '/api/v1/blueprints')
+        return [item['Blueprints']['blueprint_name'] for item in response.json()['items']]
+
+
+    def blueprint_remove(self, blueprint_name):
```
Comment (dmtucker) [inline yaas/client.py:54]: Boolean?

File: yaas/objects.py [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+
+    @classmethod
+    def parse(cls, raw):
+        ret = cls()
+        for config_block in raw:
+            for name, configs in config_block.items():
+                ret[name].update(configs)
+        return ret
+
+    def serialize(self):
```
Comment (dmtucker) [inline yaas/objects.py:20]: Could this be:

``` python
return self.items()
```

or (not sure if that's a generator)

``` python
return [item for item in self.items()]
```

?

File: yaas/objects.py [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+        if self.kerberos_descriptor is not None:
+            ret['kerberos_descriptor'] = self.kerberos_descriptor
+
+        if self.kerberos_descriptor_reference is not None:
+            ret['kerberos_descriptor_reference'] = self.kerberos_descriptor_reference
+
+        return ret
+
+class Blueprint(object):
+    # pylint: disable=too-many-arguments
```
Comment (dmtucker) [inline yaas/objects.py]: Maybe put this at the end of 95? Otherwise it will disable this check from here down I think. (Was that your intent?)

**Comment by dmtucker** [comment]
LGTM!

File: yaas/client.py (hunk 1 above)
Comment (karth295) [inline yaas/client.py:42]: Offline: decided no.

File: yaas/client.py (hunk 2 above)
Comment (karth295) [inline yaas/client.py:54]: Offline: decided no.

File: yaas/objects.py (hunk 3 above)
Comment (karth295) [inline yaas/objects.py:20]: Offline: nope.

File: yaas/objects.py (hunk 4 above)
Comment (karth295) [inline yaas/objects.py]: Fixed.

**Pull request merged**